internal/cmdaddsubpkg/cmdaddsubpkg.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmdaddsubpkg
+
+import (
+	docs "github.com/GoogleContainerTools/kpt/internal/docs/generated/pkgdocs"
+	"github.com/GoogleContainerTools/kpt/internal/kptfile"
+	"github.com/GoogleContainerTools/kpt/internal/util/cmdutil"
+	"github.com/GoogleContainerTools/kpt/internal/util/parse"
+	"github.com/GoogleContainerTools/kpt/internal/util/sync"
+	"github.com/spf13/cobra"
+)
+
+// NewRunner returns a command runner.
+func NewAddSubPkgRunner(parent string) *AddSubPkgRunner {
+	r := &AddSubPkgRunner{}
+	c := &cobra.Command{
+		Use:     "add-subpkg REPO_URI[.git]/PKG_PATH[@VERSION] LOCAL_DEST_DIRECTORY",
+		RunE:    r.runE,
+		Long:    docs.AddsubpkgLong,
+		Short:   docs.AddsubpkgShort,
+		Example: docs.AddsubpkgExamples,
+		Args:    cobra.ExactArgs(2),
+		PreRunE: r.preRunE,
+	}
+
+	c.Flags().StringVar(&r.MergeStrategy, "strategy", "", "update strategy to use, default: resource-merge")
+	cmdutil.FixDocs("kpt", parent, c)
+	r.Command = c
+	return r
+}
+
+func NewCommand(parent string) *cobra.Command {
+	return NewAddSubPkgRunner(parent).Command
+}
+
+type AddSubPkgRunner struct {
+	Dependency    kptfile.Dependency
```

Comment:
Is this needed here? Seems like we can just create this in `runE`

Reply:
Agree. Done.